File: token-validation-core/src/main/java/no/nav/security/token/support/core/exceptions/AnnotationRequiredException.java
package no.nav.security.token.support.core.exceptions;

import java.lang.reflect.Method;

public class AnnotationRequiredException extends RuntimeException {
    public AnnotationRequiredException(String message) {
        super(message);
    }

    public AnnotationRequiredException(Method method) {
        this("Server misconfigured - controller/method ["
                + method.getDeclaringClass().getName() + "." + method.getName()
                + "] not annotated @Unprotected, @Protected, @RequiredClaims or added to ignore list");
    }
}

File: token-validation-core/src/main/kotlin/no/nav/security/token/support/core/exceptions/AnnotationRequiredException.kt
package no.nav.security.token.support.core.exceptions

import java.lang.reflect.Method
import no.nav.security.token.support.core.validation.JwtTokenAnnotationHandler.Companion.SUPPORTED_ANNOTATIONS
class AnnotationRequiredException(message : String) : RuntimeException(message) {
    constructor(method : Method) : this("Server misconfigured - controller/method [${method.declaringClass.name}.${method.name}] not annotated with any of $SUPPORTED_ANNOTATIONS or added to ignore list")
}